stores/app.js

```diff
@@ -1,29 +1,79 @@
 import { defineStore } from 'pinia'
 import { use_vtk_store } from './vtk'
+import { use_ws_link_store } from './ws_link'
 const vtk_store = use_vtk_store()
+const ws_link_store = use_ws_link_store()
 
 export const use_app_store = defineStore('app', {
   state: () => ({
     display_menu: true,
     display_object_selector: false,
-    object_tree:
-    {
-      'items': []
-    }
+    object_tree: []
   }),
+  getters: {
+    are_textures_valid: (state) => (object_tree_index) => {
+      const textures = state.object_tree[object_tree_index].textures
+
+      for (let i = 0; i < textures.length; i++) {
+        const texture = textures[i]
+
+        if (texture.texture_name.is_valid === false || texture.texture_file_name.is_valid === false) {
+          return false
+        }
+      }
+      return true
+    }
+  },
   actions: {
-    add_object_tree_item (item) {
-      this.object_tree.items.push(item)
+    add_object_tree_item (id,
```

Comment:
Why give multiple arguments and not an object. Looks more maintainable an object